_posts/2018-09-06-stateless-microservices.md

```diff
@@ -0,0 +1,272 @@
+---
+title: Introducing stateless microservices for OpenFaaS
+description: Alex walks us through the latest addition to OpenFaaS - full support for stateless microservices giving an example of a guestbook written in Ruby with Sinatra
+# date: 2018-09-06
+date: 2018-09-04
+image: /images/stateless-microservices/mixed-services.jpg
+categories:
+  - kubernetes
+  - swarm
+  - microservices
+author_staff_member: alex
+dark_background: true
+---
+
+I want to share some news with you - we've merged and released support for stateless microservices in [OpenFaaS 0.9.0](https://github.com/openfaas/faas/releases/0.9.0). This means you can take advantage of OpenFaaS as a single pane of glass to manage your microservices and functions - from the CLI to the Prometheus metrics to the built-in auto-scaling, even scaling to zero and back again. I'll walk you through deploying a Ruby and Sinatra guestbook based by MySQL running on OpenFaaS with Kubernetes.
+
+## What's a Function?
+
+Before getting into what a "stateless microservice" is, let's go back to my definition of a "FaaS Function" as defined in my blog post: [Functions as a Service (FaaS) (Jan 2017)](https://blog.alexellis.io/functions-as-a-service/).
+
+Functions tend to invoke:
+
+* invoking short-lived functions (Lambda has a default 1-sec timeout)
+* does not publish TCP services - often accesses third-party services or resources
+* tends to be ad-hoc/event-driven such as responding to webhooks
+* should deal gracefully with spikes in traffic
+* despite name, runs on servers backed by a fluid, non-deterministic infrastructure
+* when working well makes the complex appear simple
+* has several related concerns: infrastructure management, batching, access control, definitions, scaling & dependencies
+
+Over the last year I've simplified this defintion into a number of properties.
```

Comment:
I’d consider replacing simplified with rationalised as the former brings a sense of devaluation